src/Logging/JUnit.php

```diff
@@ -0,0 +1,430 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Pest\Logging;
+
+use function class_exists;
+use DOMDocument;
+use DOMElement;
+use Exception;
+use function get_class;
+use function method_exists;
+use Pest\Concerns\TestCase;
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\ExceptionWrapper;
+use PHPUnit\Framework\SelfDescribing;
+use PHPUnit\Framework\Test;
+use PHPUnit\Framework\TestFailure;
+use PHPUnit\Framework\TestListener;
+use PHPUnit\Framework\TestSuite;
+use PHPUnit\Framework\Warning;
+use PHPUnit\Util\Filter;
+use PHPUnit\Util\Printer;
+use PHPUnit\Util\Xml;
+use ReflectionClass;
+use ReflectionException;
+use function sprintf;
+use function str_replace;
+use Throwable;
+use function trim;
+
+/**
+ * @internal This class is not covered by the backward compatibility promise for PHPUnit
+ */
+final class JUnit extends Printer implements TestListener
```

Comment:
Most of this file is a copy/paste from phpunit, there are very few changes to it. However it was not possible to just make a wrapper class, at least without a lot of reflections on top.

Reply:
Can't we copy this class with a post-script or something?

Reply:
Maybe its easier to inject the PHPUnit JUnit class into this one, and then only add the extra functionality where needed?